clients/src/main/java/org/apache/kafka/common/protocol/SecurityProtocol.java

```diff
@@ -83,6 +87,11 @@ public static SecurityProtocol forId(Short id) {
         return CODE_TO_SECURITY_PROTOCOL.get(id);
     }
 
+    /** Case insensitive lookup by protocol name */
+    public static SecurityProtocol forName(String name) {
+        return NAME_TO_SECURITY_PROTOCOL.get(name.toUpperCase());
```

Comment:
Can't you use `SecurityProtocol.valueOf` here instead of `NAME_TO_SECURITY_PROTOCOL.get`?